mobile/lib/modules/backup/views/backup_controller_page.dart

```diff
@@ -412,6 +412,7 @@ class BackupControllerPage extends HookConsumerWidget {
 
     void startBackup() {
       ref.watch(errorBackupListProvider.notifier).empty();
+      // TODO Android: Make sure background backup service is disabled & currently not running!
```

Comment:
Should we disable the background service as soon as the app is resumed? It that the mechanism "baked" into the background service implementation on the native side, such that when the app is in foreground, the background service would pause or stop running?

Do you have any good resouce to read on this?

Reply:
Whether to disable the service once the app is opened is good question. This is mostly a concurrency issue. What we must prevent is that the background service is uploading something, user opens the app and the foreground app also performs an upload (of potentially the same files). There are multiple options to consider:

1. When user opens the app: Stop the service (and kill any currently happening uploads), wait until stopped, check which assets to backup and perform upload as usual in the foreground app. Start service once app is closed ("inactive" / "paused"  state)
2. Remove foreground backup/upload (on Android) in the app entirely
3. Let the user decide whether to use either only foreground upload or use only the background service

Of these options, I think 2. is cleanest (and simplest), 1. is very difficult to get right and potentially wasteful regarding battery as uploads might be canceled and performed twice, 3. is nice as a safeguard if the background upload has issues (so manual foreground upload still works as is) and difficulty is somewhere between the other two options

> It that the mechanism "baked" into the background service implementation on the native side, such that when the app is in foreground, the background service would pause or stop running?

There is no automatic stopping (or pausing) of the background service. It runs entirely independent of the app.
Sorry, no good single resource to read (the [Android documentation on services](https://developer.android.com/guide/components/services) might even be confusing due to using the word "foreground" for services..)

Reply:
I like option 1 the best because it makes the most sense to the flow of the app. 

I'd like to retain the ability to let the user perform the backup as they like and have to choose to opt-in to run background backup so that we don't force users to a fixed workflow.

Reply:
I think we can set a flag in the persistent storage of the app that the user is using the app, background service can check that before performing any action

Reply:
if Hive works fine with concurrent access by different processes, this is good. However, it seems boxes need to be closed before opening in another process.

Reply:
It turns out, this might actually be possible. I've figured out a way to setup message passing between the two threads (foreground and background service)!
I think should be done in a follow-up PR once the background service is well tested and has a counter part on iOS.